examples/prodlda.py

```diff
@@ -0,0 +1,207 @@
+        logtheta_scale = jnp.ones((batch_docs.shape[0], hyperparams["num_topics"]))
+        logtheta = numpyro.sample(
+            "logtheta", dist.Normal(logtheta_loc, logtheta_scale).to_event(1)
+        )
+        theta = jax.nn.softmax(logtheta)
+
+        count_param = decoder(numpyro.prng_key(), theta, is_training)
+
+        total_count = batch_docs.sum(-1).max().astype(int)
+        numpyro.sample(
+            "obs", dist.Multinomial(total_count, count_param), obs=batch_docs
+        )
+
+
+def guide(docs, hyperparams, is_training=False):
+    encoder = haiku_module(
+        "encoder",
+        # use `transform_with_state` for BatchNorm
+        hk.transform_with_state(
+            HaikuEncoder(
+                hyperparams["vocab_size"],
+                hyperparams["num_topics"],
+                hyperparams["hidden"],
+                hyperparams["dropout_rate"],
+            )
+        ),
+        input_shape=(1, hyperparams["vocab_size"]),
+        apply_rng=True,
+        # to ensure proper initialisation of BatchNorm we must
+        # initialise with is_training=True
+        is_training=True,
+    )
+    with numpyro.plate(
+        "documents", docs.shape[0], subsample_size=hyperparams["batch_size"]
+    ):
+        batch_docs = numpyro.subsample(docs, event_dim=1)
+        logtheta_loc, logtheta_scale = encoder(
+            numpyro.prng_key(), batch_docs, is_training
+        )
+        numpyro.sample(
+            "logtheta", dist.Normal(logtheta_loc, logtheta_scale).to_event(1)
+        )
+
+
+def plot_word_cloud(b, ax, n):
+    indices = jnp.argsort(b)[::-1]
+    df = pd.DataFrame(indices[:100], columns=["index"])
+    words = pd.merge(df, vocab[["index", "word"]], how="left", on="index")[
+        "word"
+    ].values.tolist()
+    sizes = (b[indices[:100]] * 1000).astype(int).tolist()
+    freqs = {words[i]: sizes[i] for i in range(len(words))}
+    wc = WordCloud(background_color="white", width=800, height=500)
+    wc = wc.generate_from_frequencies(freqs)
+    ax.set_title("Topic %d" % (n + 1))
+    ax.imshow(wc, interpolation="bilinear")
+    ax.axis("off")
+
+
+if __name__ == "__main__":
+    rng_key = random.PRNGKey(0)
+    # TODO: remove data_home override
+    news = fetch_20newsgroups(subset="all", data_home="/tmp/scikit-learn-data")
+    vectorizer = CountVectorizer(max_df=0.5, min_df=20, stop_words="english")
+    docs = jnp.array(vectorizer.fit_transform(news["data"]).toarray())
+
+    vocab = pd.DataFrame(columns=["word", "index"])
+    vocab["word"] = vectorizer.get_feature_names()
+    vocab["index"] = vocab.index
+
+    print(f"Dictionary size: {len(vocab)}")
+    print(f"Corpus size: {docs.shape}")
+
+    # TODO: allow these to be overridden with command-line arguments
+    rng_key = random.PRNGKey(0)
+    num_topics = 20
+    docs = device_put(docs)
+    batch_size = 32
+    learning_rate = 1e-3
+    num_steps = 200_000
```

Comment:
I think 30,000 is enough (similar to the number in Pyro).

Reply:
I think I bumped this up to see what would happen when it wasn't learning anything in my initial testing and forgot I had done that 😳